### ISSUE TITLE
Change the way to handle known error and unknown error

### ISSUE THREAD

**Issue by 0x221A** [99designs/gqlgen#2735]
### What happened?

Currently, we have an `ErrorPresenter` to intercept the error response. But every error from the operation is wrapped with `ErrorOnPath`. This makes it hard to handle unknown errors and response raw error messages to the users.

https://github.com/99designs/gqlgen/blob/76d444c15fc1d98538648a229f41ed7b4573bad7/graphql/context_response.go#L52-L61

Another thing is if we want to strip some error out from the response eg. `sql.ErrNoRows` by returning `nil` value from `ErrorPresenter` will cause the app to panic for now. I suggest checking for `nil` value before appending the error to `c.errors`.

### versions
 - `gqlgen` v0.17.35
 - `go` v1.20.6

**Comment by samsullivan** [comment]
Agreed; I'm losing valuable error information with the way errors are being wrapped, making it hard to get value out of `SetErrorPresenter`.

Still looking at a way to solve my issue. If I can't figure it out, I'll try to spend a moment on a minimal repro example...

**Comment by samsullivan** [comment]
Nevermind, I figured it out; @0x221A, not sure if it'll help you, but the error type I needed was (as expected) just wrapped one level deep and easy to access with `errors.Unwrap(e)` :+1:

**Comment by 0x221A** [comment]
@samsullivan That's what I thought as well, but rather than wrap an error I ended up creating custom error types for the known error and panic error. And use `errors.As` to identify wether it should response back to the users.

I try to minimize which error is possible to come into the error presenter because gqlgen doesn't have a special error type for unmarshal functions and I don't want to strip it.
